nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-mongodb/src/main/java/org/nuxeo/directory/mongodb/MongoDBSession.java

```diff
@@ -164,6 +159,11 @@ protected DocumentModel createEntryWithoutReferences(Map<String, Object> fieldMa
             }
         }
 
+        // Check if the entry already exists
+        if (hasEntry(String.valueOf(fieldMap.get(idFieldName)))) {
+            throw new DirectoryException(String.format("Entry with id %s already exists", id));
```

Comment:
`id` in error message isn't the one with tenant information.

Reply:
Changed to reuse id